build.gradle.kts

```diff
@@ -0,0 +1,597 @@
+        jdk_home = System.getenv("JB_JAVA17_HOME")
+    }
+    if (jdk_home != null) {
+        if (!File(jdk_home, "lib").exists()) {
+            throw GradleException("Unable to locate JDK home folder. Detected folder is: $jdk_home")
+        } else {
+            extra["itemis.mps.gradle.ant.defaultJavaExecutable"] = File(jdk_home, "bin/java")
+        }
+    }
+}
+
+// enables https://github.com/mbeddr/mps-gradle-plugin#providing-global-defaults
+extra["itemis.mps.gradle.ant.defaultScriptArgs"] = defaultScriptArgs
+extra["itemis.mps.gradle.ant.defaultScriptClasspath"] = buildScriptClasspath
+
+fun createResolveTask(taskName: String, configurationName: String, outputDir: String) {
+    tasks.register<Sync>(taskName) {
+        from(configurations[configurationName])
+        into(file(outputDir))
+        // Strip version numbers from file names
+        rename { filename ->
+            val ra = configurations[configurationName].resolvedConfiguration.resolvedArtifacts.find { ra -> ra.file.name == filename }
+            val finalName = if (ra?.classifier != null) {
+                "${ra.name}-${ra.classifier}.${ra.extension}"
+            } else {
+                "${ra?.name}.${ra?.extension}"
+            }
+            return@rename finalName
+        }
+    }
+}
+
+createResolveTask("resolveDiagram", "diagram_lib", "code/diagram/solutions/de.itemis.mps.editor.diagram.runtime/lib")
+createResolveTask("resolveBatik", "batik", "code/batik/solutions/lib")
+createResolveTask("resolveCommons", "commons", "code/apache-commons/solutions/org.apache.commons/lib")
+createResolveTask("resolveCollections", "collections", "code/shadowmodels/solutions/de.q60.mps.collections.libs/lib")
+createResolveTask("resolveXML", "xml_lib", "code/xml/solutions/lib")
+
+tasks.register<BuildLanguages>("build_allScripts") {
+    dependsOn("downloadJbr", "resolveMps", "copyModelApi", "resolveDiagram", "resolveBatik", "resolveCommons", "resolveCollections", "resolveXML")
+    script = "$rootDir/scripts/build.xml"
+}
+
+tasks.register<Copy>("copyChangelog") {
+    from("$rootDir/code/solutions/de.itemis.mps.extensions.changelog/source_gen/de/itemis/mps/extensions/changelog")
+    into("$rootDir")
+    include("*.md")
+}
+
+tasks.register<BuildLanguages>("build_languages") {
+    dependsOn("build_allScripts")
+    script = scriptFile("languages/build.xml")
+}
+
+tasks.register<BuildLanguages>("build_tests") {
+    dependsOn("build_languages")
+    description = "Will build all tests without executing them."
+    script = scriptFile("tests/build.xml")
+}
+
+tasks.register("failOnTestError") {
+    description = "evaluate junit result and fail on error"
+    doLast {
+        val juniXml = file("TESTS-TestSuites.xml")
+        if (juniXml.exists()) {
+            val junitResult = XmlSlurper().parse(juniXml)
+            val failures = junitResult.depthFirst().asSequence().filter { (it as groovy.xml.slurpersupport.GPathResult).name() == "failure" }
```

Comment:
can you add GPathResult to the imports, so both lines here can be shortened?

Reply:
Done.